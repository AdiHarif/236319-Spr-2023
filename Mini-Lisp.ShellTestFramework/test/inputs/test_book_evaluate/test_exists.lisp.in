; =================================================
; exists - Check if value in list:
; =================================================
(exists 'a '())    ; NIL
(exists 'a '(x y)) ; NIL
(exists 'a '(a b)) ; T

; =================================================
; exists - Error:
; =================================================
(exists 'a 'a)     ; Error
