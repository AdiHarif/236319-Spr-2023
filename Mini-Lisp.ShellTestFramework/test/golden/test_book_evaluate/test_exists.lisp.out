NIL
NIL
T
Traceback (most recent call last):
	From function EXISTS:
		Args: (A A)
	From function COND:
		Args: (((NULL XS) XS) ((EQ X (CAR XS)) T) (T (EXISTS X (CDR XS))))
	From function EQ:
		Args: X
		Args: (CAR XS)
	From function CAR:
		Args: A
Error [CAR] in A
